routes/api/categories.js

```diff
@@ -35,6 +35,16 @@ router.get('/:id', async (req, res) => {
   }
 });
 
+router.get('/:slug/resources', async (req, res) => {
+  const record = await models.Category.findOne({ where: { slug: req.params.slug } });
+  const secondRecord = await models.Resource.findAll({ where: { CategoryId: record.id } });
```

Comment:
You're fetching the resources, but not doing anything with it. So, this API call is poorly named because it's not actually returning resources.

As I've mentioned in other comments, as an API call to just fetch a Category by slug, it's kind of redundant. The get('/:id') route can be used for this- we just need to check if the id is a number (findByPk) or text (findOne({where...}))...

One way to check is by using a regular expression... `/^\d+$/` is a regular expression that can be used to check if a string is composed of only numeric digits (which would be a primary key).

Reply:
This made a lot more sense now. Thank you for clearing it up. I have removed the extra `router.get('/:slug/resources')` function and checked for the slug in `router.get(/:id)`.